unicodetools/src/main/java/org/unicode/text/UCA/UCA_Data.java

```diff
@@ -32,21 +31,27 @@ public class UCA_Data {
 
     private final Normalizer toD;
     private final UCD ucd;
-    private final Remap primaryRemap;
 
     public int variableLow = '\uFFFF';
     public int nonVariableLow = '\uFFFF'; // HACK '\u089A';
     public int variableHigh = '\u0000';
+    boolean hasExplicitVariableHigh = false;
```

Comment:
FYI: This being true replaces a test for primaryRemap!=null. It's true for CLDR where the caller provides the variableHigh on the last punctuation character, as opposed to false for the DUCET, where the allkeys.txt parser figures it out from the data.